cpp/src/arrow/csv/writer_test.cc

```diff
@@ -233,7 +241,16 @@ std::vector<WriterTestParams> GenerateTestCases() {
       reject_structural_params({"0123456789", nullptr, "abcde,", nullptr}, "abcde,"),
       reject_structural_params({"0123456789", nullptr, "abcdef,", nullptr}, "abcdef,"),
       reject_structural_params({nullptr, nullptr, ",0123456789", "abcde"}, ",0123456789"),
-      reject_structural_params({"0123456", nullptr, "7\\\"89", ",abcdef"}, "7\"89")};
+      reject_structural_params({"0123456", nullptr, "7\\\"89", ",abcdef"}, "7\"89"),
+      // exercise custom delimiter
+      {schema_custom_delimiter, batch_custom_delimiter,
+       DefaultTestOptions(/*include_header=*/false, /*null_string=*/"",
+                          QuotingStyle::Needed, "\n", /*delimiter=*/'\t'),
+       expected_output_delimiter_tabs},
+      {schema_custom_delimiter, batch_custom_delimiter,
+       DefaultTestOptions(/*include_header=*/false, /*null_string=*/"",
+                          QuotingStyle::Needed, "\n", /*delimiter=*/'|'),
+       expected_output_delimiter_pipe}};
```

Comment:
Can we add a couple cases with invalid parameters to make sure they're properly rejected? See how reject_structural_params sets things up.

Reply:
Made the changes.